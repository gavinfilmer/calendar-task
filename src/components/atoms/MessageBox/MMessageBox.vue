<template>
  <div class="message-box" role="status">
    <div
      :class="`message-box__container ${alertTypeClass}`"
      :data-qa-locator="dataQaLocator"
    >
      <div class="message-box__icon"></div>
      <div class="message-box__content">
        <slot />
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'MMessageBox',
  props: {
    alertType: {
      type: String,
      required: false,
      default: 'info',
      validator(val) {
        return ['', 'info', 'warning', 'error'].includes(val)
      },
    },
    dataQaLocator: {
      type: String,
      default: null,
    },
  },
  computed: {
    alertTypeClass() {
      return this.alertType
        ? `message-box__container--${this.alertType}`
        : 'message-box__container--info'
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../../../sass/framework/framework.scss';

.message-box {
  color: $color-sailor;
  display: inline-block;

  &__container {
    align-items: center;
    display: flex;
    padding: 15px;

    &--info {
      background-color: $color-barely;
    }
    &--warning {
      background-color: $color-rarely;
    }
    &--error {
      background-color: $color-fairly;
    }

    // helps center a message from prismic if only 1 element
    /* stylelint-disable-next-line selector-pseudo-element-no-unknown */
    ::v-deep p:last-of-type {
      margin-bottom: 0;
    }
  }

  &__content {
    @include font-size('sm');
    margin: 0 10px;
  }

  &__icon {
    background-image: url('#{$patterns-asset-path}/svg/exclamation-circle.svg');
    background-position: center center;
    background-repeat: no-repeat;
    background-size: contain;
    margin: 0 10px;
    min-height: 30px;
    min-width: 30px;
  }
}
</style>
