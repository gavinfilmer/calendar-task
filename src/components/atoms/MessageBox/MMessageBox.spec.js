import { mount, shallowMount } from '@vue/test-utils'
import MMessageBox from './MMessageBox'

describe('MMessageBox', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(MMessageBox)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('it renders correctly', () => {
    const wrapper = mount(MMessageBox, {
      propsData: {
        dataQaLocator: 'qa-loc',
      },
      slots: {
        default: '<div>This is content</div>',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  test.each([
    [null, 'info'],
    ['', 'info'],
    ['info', 'info'],
    ['warning', 'warning'],
    ['error', 'error'],
  ])(
    'when alertType is "%s", it creates .message-box__container--%s',
    (alertType, expectedClassModifier) => {
      const alertClass = `.message-box__container--${expectedClassModifier}`
      const wrapper = shallowMount(MMessageBox, {
        propsData: {
          alertType,
          dataQaLocator: 'qa-loc',
        },
        slots: {
          default: '<div>This is content</div>',
        },
      })
      expect(wrapper.props().alertType).toBe(alertType)
      expect(wrapper.contains(alertClass)).toBe(true)
    }
  )
})
