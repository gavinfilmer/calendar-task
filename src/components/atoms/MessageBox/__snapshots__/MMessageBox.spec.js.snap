// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MMessageBox it renders correctly 1`] = `
<div role="status" class="message-box">
  <div data-qa-locator="qa-loc" class="message-box__container message-box__container--info">
    <div class="message-box__icon"></div>
    <div class="message-box__content">
      <div>This is content</div>
    </div>
  </div>
</div>
`;
